File: src/main/java/tk/hiddenname/smarthome/model/signal/DigitalState.java
package tk.hiddenname.smarthome.model.signal;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@EqualsAndHashCode(callSuper = true)
@ToString(of = {"digitalState"}, callSuper = true)
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class DigitalState extends Signal {

    public boolean isDigitalState() {
        return digitalState;
    }

    public boolean digitalState;

    public DigitalState(Long id, boolean digitalState) {
        super(id);
        this.digitalState = digitalState;
    }
}

File: src/main/kotlin/tk/hiddenname/smarthome/model/signal/DigitalState.kt
package tk.hiddenname.smarthome.model.signal

import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import javax.validation.constraints.NotNull

@JsonNaming(SnakeCaseStrategy::class)
@JsonPropertyOrder(value = ["hardwareId", "digitalState"])
class DigitalState(
        @field:NotNull(message = "digital state should be specified")
        val digitalState: Boolean?
) : Signal() {

    constructor (hardwareId: Long, digitalState: Boolean?) : this(digitalState) {
        super.hardwareId = hardwareId
    }
}